### PR TITLE
Allow usage of native currency as buy token in limit orders

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Before we disabled usage of native currency at all. But, there is no reason to ban native currency to use as a buy token.
In this PR we allow selecting of native currency as buy token.

When a user tries to set native currency as a sell token in the URL, we replace it with a wrapped token.
Also, on switching currencies places we replace native currency by wrapped.

### CODE REVIEW

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @shoom3301 , great job!
The only thing that I have to mention is that Unwrap transaction behaves like a limit order (fails, of cause). It would be nice to disable it on the form. 
wrap as a order

File: src/cow-react/modules/limitOrders/hooks/useDisableNativeTokenUsage.ts [hunk 1]
```diff
@@ -4,10 +4,11 @@ import { limitOrdersAtom } from '@cow/modules/limitOrders/state/limitOrdersAtom'
 import { useEffect } from 'react'
 import { NATIVE_CURRENCY_BUY_TOKEN } from 'constants/index'
 import { useTradeNavigate } from '@cow/modules/trade/hooks/useTradeNavigate'
+import { getDefaultTradeState } from '@cow/modules/trade/types/TradeState'
 
 /**
  * To avoid WRAP/UNWRAP/ETH-flow cases in the Limit orders page
- * We disable selecting of native currency
+ * We disable selecting of native currency as input currency
  * We set `disableNonToken` as true to `CurrencyInputPanel` component as well
```
Comment (alfetopito) [inline src/cow-react/modules/limitOrders/hooks/useDisableNativeTokenUsage.ts:12]: This is not longer true :)

Comment (shoom3301) [reply src/cow-react/modules/limitOrders/hooks/useDisableNativeTokenUsage.ts:12]: we set it only for input currency, thanks! Will update

**Comment by shoom3301** [comment]
@elena-zh Unwrap operation supposed to be implemented in another task, could you create a ticket please?

**Review by elena-zh — APPROVED** [review]
#1431 issue is created for the Unwrap case

**Review by alfetopito — APPROVED** [review]

File: src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts [hunk 2]
```diff
@@ -14,16 +14,12 @@ type PriceResult = number | Error | undefined
 async function requestPriceForCurrency(chainId: number | undefined, currency: Currency | null): Promise<PriceResult> {
```
Comment (nenadV91) [inline src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts:14]: Just a suggestion if you think it improves readability, we could do something like this
Also in both scenarios we would return the price as 1 if its **not native** and there is no currencyAddress but I'm not sure that is safe, but then again I cant think of a scenario where the currency exists, its not native and it doesn't have an address.

```javascript
async function requestPriceForCurrency(chainId: number | undefined, currency: Currency | null): Promise<PriceResult> {
  const currencyAddress = getAddress(currency)
  const parsePrice = (price: number, currency: Currency) => price * 10 ** (DEFAULT_DECIMALS + getDecimals(currency))

  if (!chainId || !currency) {
    return
  }

  try {
    if (currency.isNative || !currencyAddress) {
      return parsePrice(1, currency)
    }

    const result = await getNativePrice(chainId, currencyAddress)

    if (!result) {
      throw new Error('Cannot parse initial price')
    }

    const price = parsePrice(result.price, currency)
    if (!price) {
      throw new Error('Cannot parse initial price')
    }

    return price
  } catch (error) {
    return error
  }
}
```

Comment (shoom3301) [reply src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts:14]: Looks very beautiful! Thanks @nenadV91

**Pull request merged**